File: C#/67_AddBinary.cs
public class Solution {
    public string AddBinary(string a, string b) {
        bool carry = false;
        string ans = "";
        int i = a.Length - 1, j = b.Length - 1;   
        while (i >= 0 || j >= 0){
            bool aa = i >= 0 && a[i] == '1'? true:false;
            bool bb = j >= 0 && b[j] == '1'? true:false;
            ans += (aa ^ bb ^ carry)? '1' : '0';
            carry = (aa && bb) || (aa && carry) || (bb && carry);
            i--; j--;
        }
        if (carry) ans += "1";
        char[] arr = ans.ToArray();
        Array.Reverse(arr);        
        return new string(arr);
    }
}

File: Python/67_AddBinary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        #31
        i = len(a) - 1
        j = len(b) - 1
        ans = []
        carry = False     
        while i >= 0 or j >= 0 :
            aa = True if i >= 0 and a[i] == '1' else False
            bb = True if j >= 0 and b[j] == '1' else False
            ans.append( "1" if aa ^ bb ^ carry else "0")
            carry = aa and bb or bb and carry or aa and carry
            i -= 1
            j -= 1
        if carry: ans.append('1')
        ans.reverse()
        return ''.join(str(e) for e in ans)
            